docs/0002-fixed-length-n-tuple-trees.md

```diff
@@ -0,0 +1,223 @@
+# OCFL Community Extension 0002: Fixed Length N-tuple Trees for OCFL Storage Hierarchies
+
+  * Authors: Neil Jefferies
+  * Minimum OCFL Version: 1.0
+  * Obsoletes: n/a
+  * Obsoleted by: n/a
+
+## Overview
+
+This extension provides a general mechanism for describing a set of algorithms for mapping fixed length alphanumeric 
+\[0-9,A-Z,a-z\] identifiers onto a tree structure for constructing an OCFL Storage Hierarchy in a way that maintains 
+performance. The aim is to balance the problem of having too many files or subdirectories in any one directory with 
+having an overly deeply nested set of subdirectories. As both identifier formats and specific filesystem instances 
+may vary there is not a single optimal approach.       
+
+## Parameters
+
+* name: identifierLength
+  * description: The number of meaningful characters in the base identifier 
+  * type: integer
+  * range: 1,255
+  * default:
+* name: caseMapping
+  * description: Indicates how case in the source identifier should be handled when mapping to a path
+  * type: enumerated
+  * range: "toUpper","toLower","literal"
+  * default:
+* name: invertMapping
+  * description: Indicate if mapping should begin at the rightmost (least significant) character of the stripped identifier
+  * type: boolean
+  * default: flase
+* name: tupleSize
+  * description: Indicates the size of the chunks (in characters) that the identifier is split into during mapping
+  * type: integer
+  * range: 0,32
+  * default: 2
+* name: numberOfTuples
+  * description: Indicates how many chunks are used for path generation
+  * type: integer
+  * range: 0,32
+  * default:
+* name: shortObjectRoot 
+  * description: When true, indicates that the OCFL object root directory name should contain the remainder of the identifier not used in the n-tuples
+  * type: boolean
+  * default: false
+
+## Detailed explanation
+
+The approach described here is a generalization of the [PairTree](https://tools.ietf.org/html/draft-kunze-pairtree-01) 
+algorithm applied to fixed length identifiers. It is designed to be more flexible to reflect developments in file storage
```

Comment:
The fixed length case works for UUIDs or hashed ids but otherwise seems quite a limiting restriction on  https://tools.ietf.org/html/draft-kunze-pairtree-01

Reply:
Correct. Hence the name.